src/lib.rs

```diff
@@ -1359,7 +1365,7 @@ mod sync {
     #[cfg(feature = "portable-atomic")]
     pub(super) use portable_atomic_util::Arc;
 
-    #[cfg(all(feature = "std", not(loom)))]
+    #[cfg(all(feature = "std", not(feature = "critical-section"), not(loom)))]
```

Comment:
why wasn't the `critical_section::Mutex` import put here, too?

Reply:
Because `critical-section` doesn't rely on `loom` intrinsically.

Reply:
hm `std` also doesn't rely on `loom` intrinsically; or how is that meant?/ what does that mean?

I just thought it should belong here because that part does accumulate definitions which are basically "alternatives" between different features used throughout the code, and the underlying `Mutex` impl imo is one of those "switchable" things (unfortunately not with a 1-to-1 common/shared sufficient API)

Reply:
With std you need to replace std's primitives with loom, with critical-section you can just use a CS based on Loom.